mdbenchmark/tests/test_plot.py

```diff
@@ -0,0 +1,234 @@
+                 'All hosts will be plotted.\n' \
+                 'A total of 2 runs will be plotted.\n' \
+                 'Your file was saved as test.{} in the working directory.\n'.format(output_type)
+
+        result = cli_runner.invoke(cli.cli, [
+            'plot',
+            '--csv={}'.format(data['test.csv']), '--output-name=testfile', '--output-type={}'.format(output_type)
+        ])
+        assert result.exit_code == 0
+        assert os.path.exists("testfile.{}".format(output_type))
+
+
+@pytest.mark.parametrize("gpu,cpu", [
+    (True, True),
+    (True, False),
+    (False, True),
+])
+def test_plot_filter_dataframe_for_plotting_gpu_and_cpu(cli_runner, tmpdir, data, gpu, cpu):
+    """ check whether output types are constructed correctly
+    """
+    with tmpdir.as_cwd():
+
+        input_df = pd.read_csv(data['testcsv.csv'])
+
+        test_df = input_df
+
+        if gpu and not cpu:
+            input_df = input_df[input_df.gpu]
+        elif not gpu and cpu:
+            input_df = input_df[~input_df.gpu]
+        elif not gpu and not cpu:
+            input_df = pd.read_csv(data['empty_df.csv'])
+
+        real_df = plot.filter_dataframe_for_plotting(df=input_df, host_name=(), module_name=(), gpu=gpu, cpu=cpu)
+
+        # here we compare the dataframes for any differences
+        assert_frame_equal(input_df, real_df)
+
+
+@pytest.mark.parametrize("gpu,cpu", [
+    (False, False)
+])
+def test_plot_filter_dataframe_for_plotting_gpu_and_cpu_fail(capsys, cli_runner, tmpdir, data, gpu, cpu):
+    """ check whether output types are constructed correctly
+    """
+    with tmpdir.as_cwd():
+        input_df = pd.read_csv(data['testcsv.csv'])
+
+        expected_output = "CPU and GPU not set. Nothing to plot. Exiting."
+        plot.filter_dataframe_for_plotting(df=input_df, host_name=(), module_name=(), gpu=gpu, cpu=cpu)
+        out, err = capsys.readouterr()
+        assert out == expected_output
+
+@pytest.mark.parametrize('module_name', [
+    ('namd/2.12'),
+    ('gromacs/5.1.4-plumed2.3'),
+    ('gromacs'),
+    ('namd')
+])
+def test_plot_filter_dataframe_for_plotting_module_name(cli_runner, tmpdir, data, module_name):
+    """ check whether output types are constructed correctly
+    """
+    with tmpdir.as_cwd():
+
+        expected_df = pd.read_csv(data['testcsv.csv'])
+        expected_df = expected_df[expected_df['module'].str.contains(module_name)]
+
+        input_df = pd.read_csv(data['testcsv.csv'])
+
+        module = (module_name,)
+        real_df = plot.filter_dataframe_for_plotting(df=input_df, host_name=(), module_name=module, gpu=True, cpu=True)
+
+        assert_frame_equal(expected_df, real_df)
+
+
+@pytest.mark.parametrize("host_name", [
+    ('phys'),
+    ('draco')
+])
+def test_plot_filter_dataframe_for_plotting_host_name(cli_runner, tmpdir, data, host_name):
+    """ check whether output types are constructed correctly
+    """
+    with tmpdir.as_cwd():
```

Comment:
why are you running this in a temporary directory? are you writing files?

Reply:
yes, I am writing the pdf/png file which I also check for at the end of the tests